pkg/loop/internal/relayer/pluginprovider/ext/ccip/test/exec_provider_test.go

```diff
@@ -31,3 +38,68 @@ func TestStaticExecProvider(t *testing.T) {
 		ExecutionProvider.AssertEqual(ctx, t, ExecutionProvider)
 	})
 }
+
+func TestExecProviderGRPC(t *testing.T) {
+	t.Parallel()
+	ctx := tests.Context(t)
+
+	grpcScaffold := looptest.NewGRPCScaffold(t, setupExecProviderServer, ccip.NewExecProviderClient)
+	t.Cleanup(grpcScaffold.Close)
+	roundTripExecProviderTests(ctx, t, grpcScaffold.Client())
+}
+
+func roundTripExecProviderTests(ctx context.Context, t *testing.T, client types.CCIPExecProvider) {
```

Comment:
Contexts should be derived from the most narrowly scoped `*testing.T`. Otherwise they aren't cancelled until some ancestor test ends, at which point any goroutines leftover running from a sub-test have leaked.
```suggestion
func roundTripExecProviderTests(t *testing.T, client types.CCIPExecProvider) {
```